src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -22,11 +25,20 @@ class LeonardoService(gdDAO: DataprocDAO, dbRef: DbReference)(implicit val execu
   }
 
   def createCluster(googleProject: String, clusterName: String, clusterRequest: ClusterRequest): Future[Cluster] = {
-    gdDAO.createCluster(googleProject, clusterName, clusterRequest) flatMap { clusterResponse: ClusterResponse =>
-      dbRef.inTransaction { dataAccess =>
-        dataAccess.clusterQuery.save(Cluster(clusterRequest, clusterResponse))
+    def create() =
+      gdDAO.createCluster(googleProject, clusterName, clusterRequest) flatMap { clusterResponse: ClusterResponse =>
+        dbRef.inTransaction { dataAccess =>
+          dataAccess.clusterQuery.save(Cluster(clusterRequest, clusterResponse))
+        }
       }
+
+    dbRef.inTransaction { dataAccess =>
+      dataAccess.clusterQuery.getByName(googleProject, clusterName)
+    } flatMap {
+      case Some(c: Cluster) => throw ClusterAlreadyExistsException(googleProject, clusterName)
```

Comment:
I think this is a compiler warning because `c` is not used. Maybe `case Some(_) => throw ClusterAlreadyExistsException(googleProject, clusterName)`